packages/flutter/lib/src/material/ink_decoration.dart

```diff
@@ -259,7 +259,7 @@ class _InkState extends State<Ink> {
       _ink.decoration = widget.decoration;
       _ink.configuration = createLocalImageConfiguration(context);
     }
-    Widget current = widget.child;
+    Widget current = widget.child ?? Container();
```

Comment:
Since we might build a Padding widget, and Padding allows a null child, you could defer creating the Container until the return statement below.